src/TestInfrastructure/VsObjectModelMocks/ProjectMock.cs

```diff
@@ -129,12 +130,7 @@ string Project.Kind
 
         ProjectItems Project.ProjectItems
         {
-            get
-            {
-                ProjectItems projectItems = new ProjectItemsMock(this);
-                this.Files.Keys.ToList().ForEach(f => projectItems.AddFromFile(f));
-                return projectItems;
-            }
```

Comment:
This was essentially the problem before: the `ProjectItems` were recreated on each request from simple list of file names, so any property information was lost.